src/lib/kadm5/admin.h

```diff
@@ -455,16 +467,11 @@ krb5_error_code kadm5_init_krb5_context (krb5_context *);
 
 krb5_error_code kadm5_init_iprop(void *server_handle, char **db_args);
 
-/*
- * kadm5_get_principal_keys is used only by kadmin.local to extract existing
- * keys from the database without changing them.  It should never be exposed
- * to the network protocol.
- */
 kadm5_ret_t    kadm5_get_principal_keys(void *server_handle,
                                         krb5_principal principal,
-                                        krb5_keyblock **keyblocks,
-                                        int *n_keys);
-
+                                        krb5_kvno kvno,
```

Comment:
We have explicitly not committed to a stable API for kadm5; see the comment at the top of admin.h.  We were asked to install the kadmin headers even if we could not guarantee stability.  Having people come back later and say "well, now that you're installing the header, you've implicitly guaranteed API stability" will make us a lot less likely to agree to that kind of compromise in the future.  We have made changes to the API in the past, as in commit 1b8c72b7ef064eabb37d726e831b4618cb37d2c7.

I thought we had a #define in <kadmin/admin.h> so that source files could work with multiple versions of the API, but apparently that's only true for <kdb.h>.  We can add one, although the historically misleading api_version could be confusing.  We definitely need to bump the soname if we modify the API.

I don't see this function in Heimdal, so I'm not sure what you're talking about there.  I also can't find any other uses of this function outside of our source tree in a Google search, although that of course isn't conclusive.